server.js

```diff
@@ -1,23 +1,73 @@
 const express = require("express");
 const cors = require("cors");
-
+const bodyParser = require("body-parser");
+let changeID = 0;
 const app = express();
+app.use(express.json());
 
 app.use(cors());
 
+app.use(bodyParser.urlencoded({ extended: true }));
+
 const welcomeMessage = {
-  id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
+  id: 0
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", function (request, response) {
+  response.send({ messages });
+});
+
+
+app.get("/messages/search", (request, response) => {
+  console.log("search");
+
+  const term = request.query.term;
+  response.send(messages.filter((eachMessage) => eachMessage.text.includes(term)));
```

Comment:
Here instead of sending back the result from the filter method, you could assign the returned result to variable, and then write and if statement to send a message when the filter function does not return a value. Just in case you know.